Task.cpp

```diff
@@ -0,0 +1,230 @@
+#include <cassert>
+#include <iostream>
+
+class Custom_Vector
+{
+private:
+    size_t size;
+    size_t capacity;
+    int *buffer;
+
+public:
+    Custom_Vector()
+        : size(0), capacity(0), buffer(nullptr) {} // конструктор по умолчанию
+    Custom_Vector(int size)
+        : size(0),
+          capacity(size),
+          buffer(new int[capacity]) {} // конструктор с заданным размером
+    Custom_Vector(const std::initializer_list<int> list)
+        : size(list.size()), capacity(list.size()), buffer(new int[capacity])
+    {
+        for (size_t i = 0; i < size; i++)
+        {
+            buffer[i] = *(list.begin() + i);
+        }
+    } // Конструктор со списком инициализации
+    Custom_Vector(const Custom_Vector &other)
+        : size(other.size),
+          capacity(other.capacity),
+          buffer(new int[capacity])
+    {
+        for (size_t i = 0; i < size; i++)
+        {
+            buffer[i] = other.buffer[i];
+        }
+    } // Конструктор копирования
+    Custom_Vector(Custom_Vector &&other)
+        : size(other.size),
+          capacity(other.capacity),
+          buffer(new int[capacity])
+    {
+        buffer = other.buffer;
+        other.buffer = nullptr;
+        other.size = other.capacity = 0;
+    } // Конструктор перемещения
+    ~Custom_Vector() { delete[] buffer; } // Деструктор
+    size_t get_size()
+    {
+        return size;
+    } // Получение количества заполненных ячеек
+    size_t get_capacity()
+    {
+        return capacity;
+    } // Получение количества всех ячеек
+    void assign(size_t count, const int &value)
```

Comment:
The `assign` methods also already have a mistake. Investigate how they work and fix the error.